skdownscale/pointwise_models/quantile.py

```diff
@@ -0,0 +1,363 @@
+        else:
+            x_to_cdf = X
+
+        # do the final mapping
+        qt_kws = default_none_kwargs(self.qt_kwargs, copy=True)
+        if 'n_quantiles' not in qt_kws:
+            qt_kws['n_quantiles'] = len(X)
+
+        x_quantiles = quantile_transform(x_to_cdf, copy=True, **qt_kws)
+        x_qmapped = self.x_cdf_fit_.inverse_transform(x_quantiles)
+
+        # add the trend back
+        if self.detrend:
+            x_qmapped = x_trend.inverse_transform(x_qmapped)
+
+        return x_qmapped
+
+    def _more_tags(self):
+        return {'_xfail_checks': {'check_methods_subset_invariance': 'because'}}
+
+
+class QuantileMappingReressor(RegressorMixin, BaseEstimator):
+
+    _fit_attributes = ['_X_cdf', '_y_cdf']
+
+    def __init__(self, extrapolate=None, n_endpoints=10):
+        self.extrapolate = extrapolate
+        self.n_endpoints = n_endpoints
+
+        if self.n_endpoints < 2:
+            raise ValueError('Invalid number of n_endpoints, must be >= 2')
+
+    def fit(self, X, y, **kwargs):
+        X = check_array(
+            X, dtype='numeric', ensure_min_samples=2 * self.n_endpoints + 1, ensure_2d=True
+        )
+        y = check_array(
+            y, dtype='numeric', ensure_min_samples=2 * self.n_endpoints + 1, ensure_2d=False
+        )
+
+        X = check_max_features(X, n=1)
+
+        self._X_cdf = self._calc_extrapolated_cdf(X, sort=True, extrapolate=self.extrapolate)
+        self._y_cdf = self._calc_extrapolated_cdf(y, sort=True, extrapolate=self.extrapolate)
+
+        return self
+
+    def predict(self, X, **kwargs):
+
+        check_is_fitted(self, self._fit_attributes)
+        X = check_array(X, ensure_2d=True)
+
+        X = X[:, 0]
+
+        sort_inds = np.argsort(X)
+
+        X_cdf = self._calc_extrapolated_cdf(X[sort_inds], sort=False, extrapolate=self.extrapolate)
+
+        left = -np.inf if self.extrapolate in ['min', 'both'] else None
+        right = np.inf if self.extrapolate in ['max', 'both'] else None
+        X_cdf.pp[:] = np.interp(
+            X_cdf.vals, self._X_cdf.vals, self._X_cdf.pp, left=left, right=right
+        )
+
+        # Extrapolate the tails beyond 1.0 to handle "new extremes"
+        if np.isinf(X_cdf.pp).any():
+            lower_inds = np.nonzero(-np.inf == X_cdf.pp)[0]
+            upper_inds = np.nonzero(np.inf == X_cdf.pp)[0]
+            model = LinearRegression()
+            if len(lower_inds):
+                s = slice(lower_inds[-1] + 1, lower_inds[-1] + 1 + self.n_endpoints)
+                model.fit(X_cdf.pp[s].reshape(-1, 1), X_cdf.vals[s].reshape(-1, 1))
+                X_cdf.pp[lower_inds] = model.predict(X_cdf.vals[lower_inds].reshape(-1, 1))
+            if len(upper_inds):
+                s = slice(upper_inds[0] - self.n_endpoints, upper_inds[0])
+                model.fit(X_cdf.pp[s].reshape(-1, 1), X_cdf.vals[s].reshape(-1, 1))
+                X_cdf.pp[upper_inds] = model.predict(X_cdf.vals[upper_inds].reshape(-1, 1))
```

Comment:
I'm a little unclear on how this is different from what happens in lines 285-298?

Reply:
It is similar but this is only being applied to the X data in the predict method.